classes/class-liveblog-cpt.php

```diff
@@ -118,6 +119,45 @@ public static function hierarchical_posts_filter( $query ) {
 
 		return $query;
 	}
+
+	/**
+	 * Permalinks for child posts should use IDs, not slugs.
+	 *
+	 * @param string  $post_link The post's permalink.
+	 * @param WP_Post $post      The post in question.
+	 * @param bool    $leavename Whether to keep the post name.
+	 * @param bool    $sample    Is it a sample permalink.
+	 *
+	 * @return string
+	 */
+	public static function post_type_link( $post_link, $post, $leavename, $sample ) {
+		if ( self::$cpt_slug !== $post->post_type || 0 === $post->post_parent ) {
+			return $post_link;
+		}
+
+		return get_permalink( $post->post_parent ) . "$post->ID/";
+	}
+
+	/**
+	 * Modifies the rewrite rules for the live blog CPT.
+	 *
+	 * @param array $rules The Rules.
+	 *
+	 * @return array
+	 */
+	public static function rewrite_rules( $rules ) {
+
+		// Unset the broken rule.
+		unset( $rules['live-blog/(.+?)(?:/([0-9]+))?/?$'] );
+
+		// Matches live-blog/post-name/
+		$rules['live-blog/([^/]+)/?$'] = 'index.php?fte_liveblog=$matches[1]';
```

Comment:
@paulschreiber the fte_liveblog comes from our FTE theme that is re-registering the post type. If someone were to install this plugin solo, these rules will not work.

See the fte theme in our project at `includes/cpt/class-fivethirtyeight-liveblog.php` 

The `register_post_type()` call in this plugin ( liveblog ) is different see `classes/class-liveblog.php` The registration call is `Liveblog_CPT::register_post_type()` This is why the rewrite rules were added to the theme.